src/api/fetch-post.ts

```diff
@@ -0,0 +1,26 @@
+import { fetchPostImage } from "./fetch-post-image";
+import { fetchUsername } from "./fetch-username";
+import { fetchUserImage } from "./fetch-user-image";
+
+export const fetchPost = async (post_id: number) => {
+    const response = await fetch(`http://localhost:3000/fetch-post/${post_id}`);
+
+    console.log(response)
+
+    if (!response.ok) {
+        throw new Error(response.statusText);
+    }
+
+    const post = await response.json();
+
+    const userData = await fetchUsername(post.user_id);
+    post.user = userData;
+
+    const userImage = await fetchUserImage(post.user_id);
+    post.user.image = userImage;
```

Comment:
Fajnie by było jakby te dane od razu zwracał GET http://localhost:3000/fetch-post/${post_id}

Może ze zdjęciami nie bardzo bo z firebase, ale username to możesz na backendzie przerobić tak, żeby enpoint 
http://localhost:3000/fetch-post/ zwracał od razu też username

Nawiasem mówiąc to te endpointy są słabo nazwane, niezgodnie ze standardem REST, ale ty się skupiasz na froncie więc wybaczalne, ale bez sensu jest robić endpoint fetch-username, lepiej od razu zwrócić całego usera, z jego id, imie nazwisko i jakie tam on ma dane